### ISSUE TITLE
GTalkSMS exits when notification icon is not activated --> make a blank notification

### ISSUE THREAD

**Issue by GoogleCodeExporter** [baluragala/gtalksms#10]
```
What steps will reproduce the problem?
1. Deactivate the status notification icon
2. Go back to the home screen

What is the expected output? What do you see instead?
I would expect GTalkSMS to continue running in the background and keep my 
connection alive. Instead I see it going offline immediately.
If I activate the icon the connection will be kept alive.

What version of the product are you using? On what operating system?
1.3.2 on Cyanogenmod (Froyo) 6.1 stable

Would like to have the program run and stay connected without the icon :)
```

Original issue reported on code.google.com by `TreeOfTh...@googlemail.com` on 28 Dec 2010 at 3:43

**Comment by GoogleCodeExporter** [comment]
```
I don't have the same issue.
When I test with my phone a Magic (with Luo's ROM base on Cyanogenmod Froyo) 
GtalkSMS still working after I press "menu" or "back".
I've test with and without the widget. In both case it's working.

How much free memory do you have ? Because status bar notification force the OS 
to keep during more time the application in memory...
Once I've got GTalkSMS launched with other app, I've got 17Mo of free memory.
```

Original comment by `Florent....@gmail.com` on 28 Dec 2010 at 6:02

**Comment by GoogleCodeExporter** [comment]
Original comment by `Florent....@gmail.com` on 28 Dec 2010 at 6:02
- Changed state: **Investigating**

**Comment by GoogleCodeExporter** [comment]
```
This is a "feature" of android - if you don't use a notification icon, gtalksms 
can't set the service to "foreground" mode (by design, foreground services 
*must* have a notification in later android versions).  So when you leave the 
screen (activity) android decides it can kill the app if necessary (eg, if it 
is under memory pressure).

Another thing we could consider is to add a notification without an icon - that 
might offer the best-of-both-worlds.
```

Original comment by `skippy.hammond@gmail.com` on 28 Dec 2010 at 10:42

**Comment by GoogleCodeExporter** [comment]
```
I only have about 3MB free so I guess that explains it.
"Notification without an icon" would make me happy :). I don't know if this is 
a feature that many people would find useful though.

It could be useful if someone stole your phone.
Also the connection is super reliable now so I don't really need an icon to 
check if I'm connected or not...

```

Original comment by `TreeOfTh...@googlemail.com` on 30 Dec 2010 at 12:35

**Comment by GoogleCodeExporter** [comment]
Original comment by `Florent....@gmail.com` on 21 Jan 2011 at 7:19
- Changed title: **GTalkSMS exits when notification icon is not activated --> make a blank notification**
- Changed state: **New**
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
Hi

Having this problem on my Samsung Galaxy S too

Would be nice with "Notification without icon" ...

Regards Bent
```

Original comment by `bent.son...@gmail.com` on 6 Apr 2011 at 7:01

**Comment by GoogleCodeExporter** [comment]
```
Blank icons added
```

Original comment by `Florent....@gmail.com` on 10 Jun 2011 at 11:07
- Changed state: **WaitingForRelease**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in v2.2.4
```

Original comment by `Florent....@gmail.com` on 13 Jun 2011 at 1:24
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
Original comment by `fschm...@gmail.com` on 25 Aug 2011 at 7:51
- Changed state: **Deployed**
